### ISSUE TITLE
Symfony built-in server does not support CORS on exceptions

### ISSUE THREAD

**Issue by zaiddabaeen** [symfony/symfony#27620]
**Symfony version(s) affected**: 4.1.0

**Description**  
It seems that Symfony's built-in server strips out the CORS headers if an exception occurs.

My CorsListener:

```
        if (!$event->isMasterRequest()) {
            return;
        }

        $response = $event->getResponse();

        // Add headers for CORS
        $response->headers->set('Access-Control-Allow-Origin', '*');
        $response->headers->set('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS, PATCH');
        $response->headers->set('Access-Control-Allow-Headers', 'Authorization, Content-Type, Accept');

        $event->setResponse($response);
```

This works fine for non-exceptions.

However on an exception, the CORS headers are never sent. I am running the following using CURL on the terminal.
Healthy response:
```
HTTP/1.1 200 OK
Host: localhost:8000
Date: Sun, 17 Jun 2018 15:07:55 +0000
Connection: close
X-Powered-By: PHP/7.2.3
Cache-Control: no-cache, private
Date: Sun, 17 Jun 2018 15:07:55 GMT
Content-Type: application/json
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS, PATCH
Access-Control-Allow-Headers: Authorization, Content-Type, Accept
```

When an exception is thrown:
```
HTTP/1.0 500
Host: localhost:8000
Date: Sun, 17 Jun 2018 15:12:12 +0000
Connection: close
X-Powered-By: PHP/7.2.3
Content-Type: text/html; charset=UTF-8
```

I even re-wrote the same CORS settings in my exception listener:

```
        $response = new JsonResponse(['error' => json_decode($error, true)], $exceptionCode);

        $response->headers->set('Access-Control-Allow-Origin', '*');
        $response->headers->set('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS, PATCH');
        $response->headers->set('Access-Control-Allow-Headers', 'Authorization, Content-Type, Accept');

        $event->setResponse($response);
```

And the headers got stripped.

Can we get the build-in server to not strip the headers on server errors?

**Comment by xabbuh** [comment]
Does this not happen when using Nginx or Apache? Are you able to create a small example application that allows to reproduce your issue?

**Comment by zaiddabaeen** [comment]
Nope, I ended up installing an nginx webserver for my local environment, and I just used bare configuration (listen, server name, php-fpm, index.php). I did not add any CORS headers configuration to nginx and it worked fine on exceptions.

I am quite sure that SF built-in server is stripping them out as I did spend 4 hours on it yesterday (I really didn't want the nginx webserver). But, I'll try to get to that tonight.
